odoo/addons/base/models/ir_actions_report.py

```diff
@@ -524,12 +524,23 @@ def _run_wkhtmltopdf(
 
         files_command_args = []
         temporary_files = []
+        temp_session = None
 
         # Passing the cookie to wkhtmltopdf in order to resolve internal links.
-        if request and request.db:
+        if request and request.db and request.session.sid:
+            # Create a temporary session which will not create device logs
+            sid = root.session_store.generate_key()
```

Comment:
Instead of calling `generate_key` and `session_class`, which are a bit technical details, can't you call `new` ?
It's doing that job for you
https://github.com/odoo/odoo/blob/c931c30dc1ceb4dc2aee08344b0b2d701df63917/odoo/tools/_vendor/sessions.py#L117-L119

```suggestion
            temp_session = root.session_store.new()
            temp_session.update({
                **request.session,
                '_trace_disable': True,
                'session_token': self.env.user._compute_session_token(temp_session.sid),
            })
            root.session_store.save(temp_session)
```